src/utils/document.js

```diff
@@ -0,0 +1,86 @@
+
+/**
+ * Generate a javascript object reflecting the updates from the document.
```

Comment:
What about adding how the object is then used:

Generate an `$set` javascript object, that can be used in update operations, with the changes and addition happened in the document since ... (since when 😄 ?)